test/uk/gov/hmrc/hec/controllers/DESControllerSpec.scala

```diff
@@ -31,12 +32,13 @@ import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.ExecutionContext.Implicits.global
 
-class DESControllerSpec extends ControllerSpec {
+class DESControllerSpec extends ControllerSpec with AuthSupport {
 
   private val mockDESService = mock[DESService]
 
   override val overrideBindings =
     List[GuiceableModule](
+      bind[AuthConnector].toInstance(mockAuthConnector),
```

Comment:
can anyone shed light on why this spec fails without this mock when I'm not injecting it in the controller?

Reply:
without this `bind` over here it'll start up an actual instance of `AuthConnector` and inject that into the controller instead of having our controller pick up the mock instance - so, when the test runs the mock won't get called cos the actual instance will be running instead. `AuthConnector` isn't something that we see directly in the code - it's part of the auth-client library behind the `authorise` call.

Reply:
but I'm not injecting `AuthenticateActions` into `DESController` and `IFController`, only into `TaxCheckController`

Reply:
just realised that `AuthConnector` isn't actually being used by this spec. Not sure on this - what error do you see?

Reply:
but also, this spec doesn't need to extend `AuthSupport` I think if it's not doing auth stuff - or did you have to add it in to make an error go away?

Reply:
yeah had to add it in to get rid of the error which was that couldn't find an `AuthConnector` to inject into the `TaxCheckController` but I don't understand why it was complaining about trying to inject it into a different unrelated controller altogether

Reply:
it's trying to find an `AuthConnector` here cos in `ControllerSpec` under the hood is it's starting the whole service. I think I might have an idea of what might be happening - the module to bind that class has probably not been enabled in `application.conf`. What happens if you stick this in `application.conf`?
```
play.modules.enabled += "uk.gov.hmrc.play.bootstrap.AuthModule"
```
You may be able to remove `AuthSupport` here and elsewhere with that.

Reply:
that worked 👍 
what does that do exactly? does it make an `AuthConnector` available?

Reply:
cool :+1:  yeah, somewhere in there it will bind an instance of something to `AuthConnector`. It's not directly under our control and we haven't had authenticated endpoints in our BE yet so it was easy to miss

Reply:
updated - https://github.com/hmrc/hec/pull/18/commits/53b4f2365959956f1fb4277f9cee2aca34591327